pkg/hns/endpoint_windows.go

```diff
@@ -40,6 +51,92 @@ func GetSandboxContainerID(containerID string, netNs string) string {
 	return containerID
 }
 
+func GenerateHnsEndpoint(epInfo *EndpointInfo, n *NetConf) *hcsshim.HNSEndpoint {
+	// run the IPAM plugin and get back the config to apply
+	hnsEndpoint, err := hcsshim.GetHNSEndpointByName(epInfo.EndpointName)
+	if hnsEndpoint != nil && err != nil {
+		if hnsEndpoint.VirtualNetwork != epInfo.NetworkId {
+			_, err = hnsEndpoint.Delete()
+			if err != nil {
+				errors.Annotatef(err, "[win-cni] Failed to delete endpoint %v, err:%v", epInfo.EndpointName, err)
```

Comment:
There are a bunch of places all over this code where errors are being created and not returned

Reply:
Why do we want to use `[win-cni]` as an error message prefix?